### DISCUSSION TITLE
IAM Role needed for Front Door

### DISCUSSION THREAD

**Discussion by orjan** [polymind-inc/acmebot#385] [Q&A] [answered]
Is there a built in Role that's sensible to assign to Front door so it can read the certificates.
If not what permissions should a custom role contain in order to being able to read the certificate?

I'm aware about that it's not the main problem for this project but I haven't found anything related to this question.

It looks like this custom role will do the trick but it can probably be improved

```
{
    "properties": {
        "roleName": "role-frontdoor",
        "description": "Custom role to be able to read certificates used by Azure Front Door",
        "assignableScopes": [
            "/subscriptions/{subscription-id}"
        ],
        "permissions": [
            {
                "actions": [],
                "notActions": [],
                "dataActions": [
                    "Microsoft.KeyVault/vaults/secrets/getSecret/action"
                ],
                "notDataActions": []
            }
        ]
    }
}
```

**Comment by shibayan** [comment, marked as answer]
Assign permissions to Front Door in RBAC by referring to the following document. Creating a custom role should not be necessary in general.
https://github.com/shibayan/keyvault-acmebot/wiki/How-to-use-in-Azure-services#azure-cdn--front-door
